app_dart/lib/src/service/github_checks_service.dart

```diff
@@ -89,19 +89,32 @@ class GithubChecksService {
       slug,
       buildPushMessage.userData['check_run_id'] as int?,
     );
-    final github.CheckRunStatus status = statusForResult(build!.status);
-    final github.CheckRunConclusion? conclusion =
+    github.CheckRunStatus status = statusForResult(build!.status);
+    github.CheckRunConclusion? conclusion =
         (buildPushMessage.build!.result != null) ? conclusionForResult(buildPushMessage.build!.result) : null;
     // Do not override url for completed status.
     final String? url = status == github.CheckRunStatus.completed ? checkRun.detailsUrl : buildPushMessage.build!.url;
     github.CheckRunOutput? output;
     // If status has completed with failure then provide more details.
     if (status == github.CheckRunStatus.completed && failedStatesSet.contains(conclusion)) {
-      final Build build =
-          await luciBuildService.getBuildById(buildPushMessage.build!.id, fields: 'id,builder,summaryMarkdown');
-      output = github.CheckRunOutput(title: checkRun.name!, summary: getGithubSummary(build.summaryMarkdown));
+      if (shouldRerun(buildPushMessage)) {
```

Comment:
Please move this logic to a separate method as this method is supposed to just update the check status.